File: Data Structures & Algorithms/two-integer-sum-ii/submission-6.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left=0
        right=len(numbers)-1

        while(left<right):
            val=numbers[left]+numbers[right]

            if val<target:
                left=left+1
                continue
            if val>target:
                right=right-1
                continue
            
            return [left+1,right+1]

File: Data Structures & Algorithms/is-palindrome/submission-6.java
class Solution {
    public boolean isPalindrome(String s) {
        char[] sChar= s.toLowerCase().toCharArray();
        StringBuilder builder = new StringBuilder(); 
        for(char c : sChar){
            if(Character.isAlphabetic(c) || Character.isDigit(c)){
                builder.append(c);
            }
        }
        
        char[] bChar = builder.toString().toCharArray();
        int left = 0;
        int right = bChar.length - 1;
        while(left<=right){
            if(bChar[left]==bChar[right]){
                left++;
                right--;
            }else{
                return false;
            }
        }
        return true;
    }
}
